proxy/nginx/lualib/router.lua

```diff
@@ -30,12 +31,21 @@ local key = prefix .. ":routes:" .. host
 
 -- Try to get target for host
 res, err = red:get(key)
+
+-- Exit if route could not be read
+if err then
+    ngx.log(ngx.ERR, "error reading route: " .. err)
+    ngx.exit(ngx.HTTP_INTERNAL_SERVER_ERROR)
+end
+
 if not res or res == ngx.null then
+    ngx.log(ngx.WARN, "no route for host " .. host)
+
     -- Construct Redis key for $wildcard
     key = prefix .. ":routes:$wildcard"
     res, err = red:get(key)
     if not res or res == ngx.null then
-        return
+        ngx.exit(ngx.HTTP_NOT_FOUND)
```

Comment:
If no route was found, return 404. I agree this may not be the best solution, but I couldn't think of anything better.